### DISCUSSION TITLE
How to get a character inserted at every line breaks? (and bonus question)

### DISCUSSION THREAD

**Discussion by Omikhleia** [sile-typesetter/sile#1602] [Q&A]
**Q1**. This is both a question and a challenge: Is there a clever way to do this kind of things, where the quotation marks are repeated as the lines are broken? I obviously suspect one could tweak the line breaking algorithm in some way...

![image](https://user-images.githubusercontent.com/18075640/201489130-c54d552f-62bd-4fda-8717-d05950527b86.png)

(Example taken from a random book. This is a somewhat "old-fashioned" way of quoting text, which tends to disappear, but the ancient typographers had some pretty cool ideas, sometimes... It adds a certain emphasis to the quote. Considering the cost of making "real font" in metal in those times, and the numbers of available `«` blocks they'd need just for this weird fanciness, it's amazing!).

**Q2**. Bonus points if the answer to Q1 also satisfies the following case...

A similar problem occurs in poetry when a verse is over-long and cannot fit the justification space (I am not sure of the English term, "runovers" or "turnovers"). In some books they just be wrapped and indented, but in others the part(s) that cannot fit are right aligned, and even possibly preceded by an opening square bracket:
```
| This verse is too long for the justification   |
|                                         [space |
```

(so first line is ragged right, extra broken lines are ragged left, with an extra bracket when the line is broken)

Wait, I'm no done... Poetry can get worse than that. Ideally, line-broken part(s) of a verse _should_ be set on the same line as the next verse if possible, or on the preceding verse if it cannot fit, say the manuals... That is:
```
| This verse is too long for the justification   |
| Next verse is short                     [space |
```
Or

```
| This verse is short                     [space |
| This verse is too long for the justification   |
| That verse cannot be used as it is too long    |
```

So here the problem becomes "harder"
- Insert something after line-breaks as above ... but in a ragged environment
- But also change justification alignment (left > right)
- Possibly using the next or previous lines,
- Only using a blank line when there is no other choice (or on the last verse, quite obviously)

Wow! Poetry is pretty hard... a challenge for would-be typographers and programmers!

**Comment by ctrlcctrlv** [comment]
Good question. It seems like the linebreaking stuff could really use some user-definable callbacks…

**Reply by Omikhleia** [reply]
> Good question. It seems like the linebreaking stuff could really use some user-definable callbacks…

... Or anything else that does the trick.

You know, I am just here maybe to prove the [Wiki](https://github.com/sile-typesetter/sile/wiki/easy-jobs) is wrong listing poetry and quotes as "easy tasks" that just need "messing with margins"...

**Reply by alerque** [reply]
That "easy jobs" list is total bunk. Half of it is implemented (easy or not) and the other half is either crazy complicated or even out of scope. I've just nuked the whole page.
